v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -186,67 +220,78 @@ private PubsubMessage publishModJsonStringToPubsubMessage(String modJsonString)
-                                                + ". Supported output formats: JSON, AVRO";
-                                LOG.info(errorMessage);
-                                throw new IllegalArgumentException(errorMessage);
-                        }
-                    } else {
-
+                    ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+                    if (!topic.getSchemaSettings().getSchema().isEmpty()){
+                        messageFormat = topic.getSchemaSettings().getSchema();
+                        messageEncoding = topic.getSchemaSettings().getEncoding().toString();
+                    }
+                    switch(messageFormat){
+                        case "AVRO":
+                            ChangelogEntry changelogEntry = mapJsonToChangelogEntry(modJsonString);
+                            Encoder encoder = null;
+                            switch(messageEncoding) {
+                                case "BINARY":
+                                    System.out.println("Preparing a BINARY encoder...");
+                                    encoder = EncoderFactory.get().directBinaryEncoder(byteStream, /*reuse=*/ null);
+                                    break;
+                                case "JSON":
+                                    System.out.println("Preparing a JSON encoder...");
+                                    encoder = EncoderFactory.get().jsonEncoder(ChangelogEntry.getClassSchema(), byteStream);
+                                    break;
+                                default:
+                                    break block;
+                            }
+                            changelogEntry.customEncode(encoder);
+                            encoder.flush();
+
+                            // Publish the encoded object as a Pub/Sub message.
+                            ByteString data = ByteString.copyFrom(byteStream.toByteArray());
+                            PubsubMessage message = PubsubMessage.newBuilder().setData(data).build();
+                            System.out.println("Publishing message: " + message);
+
+                            ApiFuture<String> future = publisher.publish(message);
+                            System.out.println("Published message ID: " + future.get());
+
+
+                        case "Protocol Buffer":
+                            ChangelogEntryProto changelogEntryProto = mapJsonToChangelogEntryProto(modJsonString);
+                            data = ByteString.copyFrom(byteStream.toByteArray());
+                            PubsubMessage.Builder protoMessage = PubsubMessage.newBuilder();
+                            switch(messageEncoding) {
+                                case "BINARY":
+                                    protoMessage.setData(changelogEntryProto.toByteString());
+                                    System.out.println("Publishing a BINARY-formatted message:\n" + protoMessage);
```

Comment:
I understand that this is used for debugging, but before publishing a new PR version, please self-review for System.out / System.err mentions.